src/components/DateTimePicker.js

```diff
@@ -0,0 +1,292 @@
+
+const StyleConstants = require('../constants/Style');
+
+const DatePicker = React.createClass({
+  propTypes: {
+    calendarStyle: React.PropTypes.object,
+    closeOnDateSelect: React.PropTypes.bool,
+    dateFormat: React.PropTypes.string,
+    dateIcon: React.PropTypes.string,
+    datePlaceholder: React.PropTypes.string,
+    locale: React.PropTypes.string,
+    minimumDate: React.PropTypes.number,
+    onDateSelect: React.PropTypes.func,
+    primaryColor: React.PropTypes.string,
+    selectedDate: React.PropTypes.number,
+    showIcons: React.PropTypes.bool,
+    styles: React.PropTypes.object,
+    timeFormat: React.PropTypes.string,
+    timeIcon: React.PropTypes.string,
+    timePlaceholder: React.PropTypes.string,
+    timezone: React.PropTypes.string,
+    timezoneFormat: React.PropTypes.oneOf(['abbr', 'name']),
+    timezoneNames: React.PropTypes.object
+  },
+
+  getDefaultProps () {
+    return {
+      closeOnDateSelect: true,
+      dateFormat: 'MMM D, YYYY',
+      dateIcon: 'calendar',
+      datePlaceholder: 'Select a Date',
+      locale: 'en',
+      onDateSelect () {},
+      primaryColor: StyleConstants.Colors.PRIMARY,
+      showIcons: true,
+      timeFormat: 'LT',
+      timeIcon: 'clock',
+      timePlaceholder: 'Select a Time',
+      timezone: moment.tz.guess(),
+      timezoneNames: {
+        EST: 'Eastern Standard Time',
+        EDT: 'Eastern Daylight Time',
+        CST: 'Central Standard Time',
+        CDT: 'Central Daylight Time',
+        MST: 'Mountain Standard Time',
+        MDT: 'Mountain Daylight Time',
+        PST: 'Pacific Standard Time',
+        PDT: 'Pacific Daylight Time'
+      }
+    };
+  },
+
+  getInitialState () {
+    return {
+      showCalendar: false,
+      editTime: false
+    };
+  },
+
+  _handleDateBlur () {
+    this.setState({
+      showCalendar: false
+    });
+  },
+
+  _handleDateFocus () {
+    this.setState({
+      showCalendar: true
+    });
+  },
+
+  _handleDateSelect (date, e) {
+    e.stopPropagation();
+
+    if (this.props.closeOnDateSelect) {
+      this.refs.dateSelect.blur();
+
+      this.setState({
+        showCalendar: false
+      });
+    }
+
+    const hour = this.props.selectedDate ? moment.unix(this.props.selectedDate).hour() : 0;
+    const minutes = this.props.selectedDate ? moment.unix(this.props.selectedDate).minute() : 0;
+
+    this.props.onDateSelect(moment.unix(date).hour(hour).minute(minutes).seconds(0).unix());
+  },
+
+  _handleTimeBlur () {
+    this.setState({
+      editTime: false
+    });
+  },
+
+  _handleTimeFocus () {
+    this.setState({
+      editTime: true
+    });
+  },
+
+  _handleTimeSelect (e) {
+    const selectedDate = this.props.selectedDate ? moment.unix(this.props.selectedDate) : moment();
+    const value = e.target.value.toLowerCase().replace('.', '');
+
+    const time = value.split(':');
+    const pm = value.includes('pm');
+    const hour = pm ? Number(time[0]) + 12 : Number(time[0]);
```

Comment:
If they enter military time when using Firefox, then it handles it correctly. This issue was entering 1:00 PM. The time split would make it 01 instead of 13. If they enter 13:00 then the split works as expected.